app/src/main/java/org/literacyapp/authentication/AuthenticationActivity.java

```diff
@@ -167,7 +168,7 @@ public Mat onCameraFrame(CameraBridgeViewBase.CvCameraViewFrame inputFrame) {
 
                         if (isFaceInsideFrame){
                             if (!recognitionThread.isAlive() && !recognitionThreadStarted){
-                                if (mediaPlayerAnimalSound != null){
+                                if (!mediaPlayerAnimalSoundReleased){
```

Comment:
It fixes the problem with the IllegalStateException which comes because the MediaPlayer was already release (in another thread) but this thread still thinks the MediaPlayer object exists (http://stackoverflow.com/questions/14710456/how-to-check-if-mediaplayer-is-released). But I just realized, that I need to change it anyway. So, that the whole thing is not executed anymore (not only the animalSound).